client/websocket_client.tsx

```diff
@@ -55,19 +65,28 @@ export default class WebSocketClient {
             console.log('websocket connecting to ' + connectionUrl); //eslint-disable-line no-console
         }
 
-        this.conn = new WebSocket(connectionUrl);
+        // Add connection id, and last_sequence_number to the query param.
+        // We cannot use a cookie because it will bleed across tabs.
+        // We cannot also send it as part of the auth_challenge, because the session cookie is already sent with the request.
+        this.conn = new WebSocket(`${connectionUrl}?connection_id=${this.connectionId}&sequence_number=${this.serverSequence}`);
         this.connectionUrl = connectionUrl;
 
+        const config = getConfig(store.getState());
+        const reliableWebSockets = config.EnableReliableWebSockets === 'true';
+
         this.conn.onopen = () => {
-            this.serverSequence = 0;
+            // No need to reset sequence number here.
+            if (!reliableWebSockets) {
+                this.serverSequence = 0;
+            }
 
             if (token) {
                 this.sendMessage('authentication_challenge', {token});
             }
 
             if (this.connectFailCount > 0) {
                 console.log('websocket re-established connection'); //eslint-disable-line no-console
-                if (this.reconnectCallback) {
+                if (!reliableWebSockets && this.reconnectCallback) {
```

Comment:
So with the reliable websocket enabled, the reconnect logic is entirely handled when we receive the HELLO request and not when the connection is opened, right?

It's absolutely not necessary, but perhaps it would make sense to do the first connection handling callback in that same case as well

Reply:
I mean that we could potentially move the `firstConnectCallback` call into the `onmessage` handler like the reconnect instead of having it done in `onopen`. That's just a thought though.

Reply:
Ah I see. I would say that it's slightly out of scope for this PR. We can do it any time later.